content/blog/2016/2016-12-16-declarative-pipeline-beta.adoc

```diff
@@ -0,0 +1,111 @@
+
+Declarative Pipeline does not replace Pipeline but extends it with a pre-defined structure to let users focus entirely on the `steps` required at each `stage` without needing to worry about scripting every aspect of the pipeline. Granular flow-control is extremely powerful and Scripted Pipeline syntax will always be part of Pipeline but it's not for everyone.
+
+Declarative Pipeline enables all users to connect simple, declarative blocks that define build agents (including Docker), post build actions, environment settings, credentials and all stages that make up the pipeline. Best of all, because this Declarative syntax is part of Pipeline, all build steps and build wrappers available in Plugins or loaded from Shared Libraries are also available as `steps` in Declarative.
+
+== Example
+
+The following Pipelines should execute the same steps. The first shows the Declarative syntax and the second shows the original Scripted syntax. The Declarative syntax has a more straightforward structure that is easier to grok by users not versed in Groovy.
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent  label:'has-docker', dockerfile: true
+    environment {
+        GIT_COMMITTER_NAME = "jenkins"
+        GIT_COMMITTER_EMAIL = "jenkins@jenkins.io"
+    }
+    post {
+        always {
+          deleteDir()
+        }
+        success {
+            mail to:"me@example.com", subject:"SUCCESS: ${currentBuild.fullDisplayName}", body: "Yay, we passed."
+        }
+        failure {
+            mail to:"me@example.com", subject:"FAILURE: ${currentBuild.fullDisplayName}", body: "Boo, we failed."
+        }
+    }
+    stages {
+        stage("Build") {
+          steps {
+            sh 'mvn clean install -Dmaven.test.failure.ignore=true'
+          }
+        }
+        stage("Archive"){
+          steps {
+            archive "*/target/**/*"
+            junit '*/target/surefire-reports/*.xml'
+          }
+        }
+    }
+}
+
+// Script //
+withEnv(["GIT_COMMITTER_NAME = jenkins","GIT_COMMITTER_EMAIL = jenkins@jenkins.io"]) {
+  node('has-docker') {
+    try {
+      checkout scm // checks out Dockerfile and source code
+      def myImage = docker.build 'my-environment:snapshot'
+      myImage.inside {
+        stage('Build') {
+          sh 'mvn clean install -Dmaven.test.failure.ignore=true'
+        }
+        stage('Archive') {
+          archive "*/target/**/*"
+          junit '*/target/surefire-reports/*.xml'
+        }
+      }
+      if (currentBuild.result == null || currentBuild.result == 'SUCCESS') {
+        mail to:"me@example.com", subject:"SUCCESS: ${currentBuild.fullDisplayName}", body: "Yay, we passed."
+      }
+    }
+    catch (exc) {
+      if (currentBuild.result == 'FAILURE') {
```

Comment:
if I do:

`echo "Current Build: $currentBuild.result"`

it reports it as "FAILURE"